src/AutoMapper/QueryableExtensions/Impl/EnumerableExpressionBinder.cs

```diff
@@ -29,17 +28,24 @@ private static MemberAssignment BindEnumerableExpression(IConfigurationProvider
                 {
                     return null;
                 }
-                expression = transformedExpressions.Aggregate(expression, (source, lambda) => Select(source, lambda));
+                expression = transformedExpressions.Aggregate(expression, Select);
             }
 
-            expression = Expression.Call(typeof(Enumerable), propertyMap.DestinationType.IsArray ? "ToArray" : "ToList", new[] { destinationListType }, expression);
+            expression =
+                propertyMap.DestinationType.IsArray
+                    ? Expression.Call(typeof(Enumerable), nameof(Enumerable.ToArray), new[] {destinationListType}, expression)
```

Comment:
Some replication here. Only the method name is different. And it's the same for IList and ICollection.

Reply:
Changed to just do `ICollection<T>` since that covers both cases